src/main/java/reposense/parser/ReportConfigYamlParser.java

```diff
@@ -0,0 +1,59 @@
+package reposense.parser;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.file.Path;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.gson.Gson;
+
+import reposense.model.reportconfig.ReportConfiguration;
+import reposense.system.LogsManager;
+
+/**
+ * YAML Parser for report-config.yaml files.
+ */
+public class ReportConfigYamlParser extends JsonParser<ReportConfiguration> {
+    public static final String REPORT_CONFIG_FILENAME = "report-config.yaml";
+    private static final Logger logger = LogsManager.getLogger(ReportConfigYamlParser.class);
+
+    @Override
+    public Type getType() {
+        return ReportConfiguration.class;
+    }
+
+    @Override
+    public ReportConfiguration parse(Path path) throws IOException {
+        return this.fromJson(null, path, null);
+    }
+
+    @Override
+    protected ReportConfiguration fromJson(Path path) throws IOException {
+        return this.fromJson(null, path, null);
+    }
+
+    @Override
+    protected ReportConfiguration fromJson(Gson gson, Path path, Type type) throws IOException , JsonMappingException {
```

Comment:
`fromYaml`?

Reply:
I initially wanted to follow the same interface as `JsonParser`, but in retrospect it might have been better to declare another abstract `YamlParser` that implements similar interface methods as `JsonParser`. I could probably work on this in the upcoming weeks if I have some time; anyone is welcome to to jump in and refactor the code if this feature is needed urgently!